src/transform/class/inheritance/index.js

```diff
@@ -0,0 +1,34 @@
+import UtilInherits from './UtilInherits';
+import Prototypal from './Prototypal';
+
+/**
+ * Processes nodes to detect super classes and return information for later
+ * transformation.
+ */
+export default class Inheritance {
+  /**
+   * @param {Object} cfg
+   *   @param {PotentialClass[]} cfg.potentialClasses Class name
+   */
+  constructor({potentialClasses}) {
+    this.potentialClasses = potentialClasses;
+    this.utilInherits = new UtilInherits(this);
+    this.prototypal = new Prototypal(this);
```

Comment:
Looking deeper, it seems that this `potentialClasses` array (hmm... it's documented as array, but it's actually a map) is used very little by both of these classes. I think the existence checking of class names doesn't actually need to happen inside these classes. We could move it completely out of there - eliminating this coupling.